cranelift/codegen/src/opts/selects.isle

```diff
@@ -98,3 +98,5 @@
         (select ty c
             (iconst ty (imm64_masked ty (u64_wrapping_add x z)))
             (iconst ty (imm64_masked ty (u64_wrapping_add y z)))))
+
+(rule (simplify (select ty d (select ty c x y) (select ty d y x))) (select ty d (select ty c x y) x))
```

Comment:
It seems like the `(select ty c x y)` (the nested select) is unimportant to the left-hand side, since it is just passed through as an opaque value / preserved as the first arg on the output?

Could we generalize this to

```
(rule (simplify (select ty d a (select ty d x y)))
                (select ty d a y))
```

in other words, "path-sensitive constant propagating" `d` to be always-false in the second select input?

Likewise for the symmetrical case, with a nested select in the first input...